src/dmd/dmangle.d

```diff
@@ -416,6 +416,34 @@ public:
         // Write argument types
         paramsToDecoBuffer(t.parameterList.parameters);
         //if (buf.data[buf.length - 1] == '@') assert(0);
+        if (t.parameterList.varargs == VarArg.variadic && t.parameterList.stc)
+        {
+            /* M Nk TypeModifiers Y
+             */
+            const stc = t.parameterList.stc;
+            if (stc & STC.scope_ && !(stc & STC.scopeinferred))
+                buf.writeByte('M');
+            if ((stc & (STC.return_ | STC.wild)) == STC.return_ &&
```

Comment:
Could you elaborate why `return` and  `inout` are not independent?

Reply:
The logic is the same as in `override void visit(Parameter p)` and the comment is:
```
// 'return inout ref' is the same as 'inout ref'
```